ProjectTemplate/Services/Network/NetworkStructs.swift

```diff
@@ -29,13 +30,56 @@ enum RequestError: Error {
     case missingRefreshToken
 }
 
+extension RequestError: ErrorPresentable {
+    var title: String? {
+        switch self {
+        case .missingRefreshToken, .network, .upload: return L10n.Basic.error
+        }
+    }
+
+    var message: String {
+        switch self {
+        case .missingRefreshToken, .upload: return L10n.Basic.Error.message
+        case .network(let e):
+            switch (e.error as NSError).code {
+            case -1001, -1009: return e.error.localizedDescription // timeout and no connection errors
+            default: return L10n.Basic.Error.message
+            }
+        }
+    }
+
+    var detailedDescription: String? {
+        switch self {
+        case .missingRefreshToken, .upload: return "\(self)"
+        case .network(let e):
+            guard let request = e.request else {
+                return "\(self)"
+            }
+
+            if let response = e.response.flatMap({ Reqres.formatResponse($0, request: e.request, method: e.request?.httpMethod, data: e.data, level: .verbose) }) {
+                return Reqres.formatRequest(request, level: .verbose) + "\n\n\n\n" + response
+            } else {
+                return Reqres.formatError(request, error: e.error as NSError)
+            }
+        }
+    }
+}
+
 struct NetworkError: Error {
     let error: Error
     let request: URLRequest?
     let response: HTTPURLResponse?
     let data: Data?
 
     var statusCode: Int? { return response?.statusCode }
+
+    var jsonBody: Any? { return data.flatMap({try? JSONSerialization.jsonObject(with: $0, options: .allowFragments) }) }
+
+    var backendStatusCode: Int? {
+        let json = jsonBody as? [String: Any]
+        let error = json?["error"] as? [String: Any]
+        return error?["code"] as? Int
+    }
```

Comment:
This is backend specific, and it's not needed for template functionality - it's not good idea to have this here.